Lib/email/utils.py

```diff
@@ -94,6 +88,8 @@ def formataddr(pair, charset='utf-8'):
             name.encode('ascii')
         except UnicodeEncodeError:
             if isinstance(charset, str):
+                # lazy import to improve module import time
+                from email.charset import Charset
```

Comment:
Not sure I like this one, only fine with it since `formataddr` doesn't look too widely used (a lot of the time it's nice to pay these costs upfront, predictable performance is important, e.g. don't want the first request your webserver serves to be randomly slow)

Reply:
> (a lot of the time it's nice to pay these costs upfront, predictable performance is important

agreed. On the other hand, though, the `email` package goes in quite heavily for lazy imports in some other places, so this does seem in keeping with that general philosophy:

https://github.com/python/cpython/blob/f7860295b16a402621e209871c8eaeeea16f464e/Lib/email/__init__.py#L28-L61

Reply:
I'd be happy to change it so it's imported at the top of the function if you think that'd be better?

Reply:
Nah that'd be worse, module level or here. I'm fine with this as is!